main/src/main/java/org/mobilitydata/gtfsvalidator/util/VersionResolver.java

```diff
@@ -0,0 +1,150 @@
+package org.mobilitydata.gtfsvalidator.util;
+
+import com.google.common.flogger.FluentLogger;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.SettableFuture;
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.Resource;
+import io.github.classgraph.ScanResult;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.jar.Manifest;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Methods to resolve the {@link VersionInfo} for the current validator instance. Since resolving
+ * the latest release version requires an external network request, we resolve the version
+ * asynchronously.
+ */
+public class VersionResolver {
+
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+
+  /** We look up the latest release version at the following wiki page. */
+  private static final String LATEST_RELEASE_VERSION_PAGE_URL =
```

Comment:
Using the wiki to retrieve the latest version number seems fragile to me - it's likely to break or be forgotten over time. Is there is a reason not to use the GitHub API to get the latest release?

https://docs.github.com/en/rest/releases/releases#get-the-latest-release

Reply:
I would note that this approach has worked for `transitfeed` for years now.

The thing I like about the wiki approach is that it allows you to control the "upgrade" notification independently of the GH Release process.  Namely, there is a world where we cut a release and let it go live for a period of time (a week?) before updating the wiki to trigger the upgrade notification.  This gives some canary time to make sure there wasn't some issue with the release with an initially smaller population before encouraging everyone to upgrade.

Reply:
Interesting, I like the idea. @isabelle-dr Do you think adding this to the release/post-release process would be helpful for the community and maintainable in the long run?

Reply:
I think this would be maintainable and I see value in what @bdferris-v2 is describing. 
The [RELEASE.md](https://github.com/MobilityData/gtfs-validator/blob/master/docs/RELEASE.md) would need to be updated to reflect this.